code/game/machinery/omnibox.dm

```diff
@@ -0,0 +1,264 @@
+			if(!active && !crit_fail)
+				active = TRUE
+				src.updateUsrDialog()
+				if(active && !crit_fail && anchored)
+					icon_state = "[src]1"
+		if("disable")
+			if (active)
+				active = FALSE
+				drain_rate = 0
+				src.updateUsrDialog()
+		if("set_power")
+			drain_rate = input("Power export rate (in kW):", name, drain_rate)
+			src.updateUsrDialog()
+			if(active && !crit_fail && anchored)
+				icon_state = "[src]1"
+		if ("close")
+			usr.unset_machine()
+		if ("Vehicle Factory")
+			var/datum/omnibox/factory/F
+			for(var/obj/machinery/MA in F.types)
+				MA = new(loc)
+			qdel(src)
+
+
+/obj/machinery/power/omnibox/process()
+	if(active && !crit_fail && anchored && powernet && drain_rate && (power_drained < goal))
+		if(powernet.netexcess >= 1)
+			powernet.load += drain_rate*1000
+			power_drained += drain_rate
+		else
+			visible_message("Power conversion levels have exceeded energy grid supply, shutting down")
+			active = FALSE
+			drain_rate = 0
+			icon_state = "[src]0"
+	else
+		active = FALSE
+		drain_rate = 0
+		icon_state = "[src]0"
+
+/datum/omnibox
+	var/id = "Placeholder"
+	var/desc = "This shouldn't exist"
+	var/list/types = list()
+
+/datum/omnibox/factory
+	id = "Vehicle Factory"
+	desc = "An advanced assembly line capable of turning raw energy into prototype vehicles"
+	types = list(
+		/obj/machinery/power/omnibox/factory,
+		/obj/machinery/power/omnibox/omnicube
+	)
+
+/obj/machinery/power/omnibox/omnicube
+	name = "Omnicube Generator"
+	desc = "Generates volatile omnicubes from the power grid that are designed for use in the vehicle factory."
+	icon_state = "Omnicube Generator0"
+	goal = 0
+	var/cube = 10000
+	var/surplus = 0
+
+/obj/machinery/power/omnibox/omnicube/Initialize()
+	. = ..()
+	optional_message = "<B>Based on current power levels, the [src] is expected to create [power_drained/10000] omnicube every two seconds.</B><br>"
+
+/obj/machinery/power/omnibox/omnicube/process()
+	if(active && !crit_fail && anchored && powernet)
+		if(powernet.netexcess >= 1)
+			powernet.load += drain_rate*1000
+			power_drained += drain_rate
+			surplus += drain_rate
+			while(surplus >= cube)
+				surplus -= 10000
+				new /obj/item/omnicube(locate(min(x+1,world.maxx),y,z))
+		else
+			visible_message("Power conversion levels have exceeded energy grid supply, shutting down")
+			active = FALSE
+			drain_rate = 0
+			icon_state = "[src]0"
+	else
+		active = FALSE
+		drain_rate = 0
+		icon_state = "[src]0"
+
+/obj/machinery/power/omnibox/factory
+	name = "Vehicle Factory"
+	desc = "A quantum-calibrated factory that consumes omnicubes in order to generate unique prototype vehicles."
+	icon_state = "Vehicle Factory"
+	var/finished = 60
+	var/progress = 0
+	var/reset = 1800
+	var/iterations = 0
+	var/record = 0
+	var/efficiency = 0.2
+	var/ideal
+	var/phase1
+	var/phase2
+	var/phase3
+
+
+/obj/machinery/power/omnibox/factory/Initialize()
+	. = ..()
+	phase1 = rand(10)
+	phase2 = rand(10,15)
+	phase3 = rand(15,20)
```

Comment:
you can potentially have 10 as p1 and p2 and 15 as both p2 and p3.

Reply:
Good point, ill adjust it